lib/iris/tests/unit/coords/test_Coord.py

```diff
@@ -303,6 +303,30 @@ def test_numeric_nd(self):
                                                                 [4, 10],
                                                                 [5, 11]]))
 
+    def test_numeric_nd_bounds(self):
+        self.setupTestArrays((3, 4))
+        coord = AuxCoord(self.pts_real, bounds=self.bds_real)
+
+        collapsed_coord = coord.collapsed()
+        self.assertArrayEqual(collapsed_coord.points, np.array([55]))
+        self.assertArrayEqual(collapsed_coord.bounds, np.array([[-2, 112]]))
+
+        # Test partially collapsing one dimension...
+        collapsed_coord = coord.collapsed(1)
+        self.assertArrayEqual(collapsed_coord.points, np.array([15,  55,  95]))
+        self.assertArrayEqual(collapsed_coord.bounds, np.array([[-2,  32],
+                                                                [38,  72],
+                                                                [78, 112]]))
+
+        # ... and the other
+        collapsed_coord = coord.collapsed(0)
+        self.assertArrayEqual(
+            collapsed_coord.points, np.array([40, 50, 60, 70]))
+        self.assertArrayEqual(collapsed_coord.bounds, np.array([[-2, 82],
+                                                                [8,  92],
+                                                                [18, 102],
+                                                                [28, 112]]))
```

Comment:
Could you split this into 3 tests? i.e. one test for each collapse operation.

My concern with having them all in the one test is that if the first one fails we won't know the result of the other two collapses until we get the first one working.
I realise `test_numeric_nd()` that already exists in this file does the same thing as you have done here, but I do think it's better practice to split the tests up

Reply:
And perhaps add a test where the dimensions given are negative to test that handling is correct:
i.e. 
```
collapsed_coord = coord.collapsed(-1)
```

Reply:
Will do, and thanks for explaining why - it makes sense.  Should I do the same with the lazy equivalent below?

Reply:
> Should I do the same with the lazy equivalent below?

Yes please!

Reply:
Done.  Also moved the dask import to the top of the module.  This is consistent with other test modules that use dask.

Since I now have `self.setupTestArrays((3, 4))` in 8 of the tests, I wondered about adding a setUp method to contain that.  But there are a lot of tests on the class that don't need these arrays so not really sure what the better approach is.